xc/xc7/yosys/utils.tcl

```diff
@@ -0,0 +1,22 @@
+# Update the CLKOUT[0-5]_PHASE parameter values.
+# Due to the fact that Yosys doesn't support floating parameter values
+# i.e. treats them as strings, the parameter values need to be multiplied by 1000
+# for the PLL registers to have correct values calculated during techmapping.
+proc update_pll_params {} {
+    set temp_file "[pid].tmp"
+    select t:PLLE2_ADV -list -write $temp_file
+    set fh [open $temp_file r]
+    foreach cell [gets $fh] {
+	for {set output 0} {$output < 6} {incr output} {
+	    set param_name "CLKOUT${output}_PHASE"
+	    set param_value [getparam $param_name t:PLLE2_ADV]
```

Comment:
Shouldn't you select the cell by its name here?

Reply:
`t:PLLE2_ADV` selects all cells of type PLLE2_ADV. Following Yosys's doc:
`t:<pattern>
        all cells with a type matching the given pattern
`

Reply:
Yeah, that is correct. But what if there is more than one PLLE2_ADV cell in the design? Which one you will retrieve the parameter from?

Reply:
Oh, now I see what you meant. I thought you were referring to `select` not `getparam`. 
Of course, that's a bug, should be `$cell` instead. Thanks.